client/src/views/TournamentView.tsx

```diff
@@ -1,86 +1,86 @@
-import React, { ReactNode } from 'react'
+import React, { useState, useEffect } from 'react'
 import { Tournament } from '../../../src/season/tournament'
 import axios from 'axios'
 import { TournamentRow, TournamentStatus } from '../components/TournamentRow'
-import { Accordion, Card } from 'react-bootstrap'
+import { ExpansionPanel, ExpansionPanelSummary, ExpansionPanelDetails } from '@material-ui/core'
+import { makeStyles } from '@material-ui/styles'
+import Typography from '@material-ui/core/Typography'
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
 
-export interface TournamentViewState {
-  tournaments: Tournament[]
-}
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: '100%',
+  },
+  heading: {},
+}))
 
-export class TournamentView extends React.Component<{}, TournamentViewState> {
-  constructor(props: object) {
-    super(props)
-    this.state = { tournaments: [] }
-  }
+export function TournamentView(): JSX.Element {
+  const [tournaments, setTournaments] = useState([] as Array<Tournament>)
```

Comment:
For `useState` it's better to write it like:

```ts
const [tournaments, setTournaments] = useState<Tournament[]>([])
```

`Tournament[]` is 100% equivalent to `Array<Tournament>`, but it's a bit shorter, and IMO easier to read cause it avoids the mess of angle brackets, specially when types get more complex.

Now about `useState<Type>` instead of `useState(variable as Type)`:

The first style instructs Typescript that `useState` expects `Type`, the second style cheats `useState` by saying that what was passed is `Type` regardless of it being or not `Type`. This cheating makes it easier to introduce bugs, so it's better to use style 1.